modules/vaos/app/services/vaos/v2/appointments_service.rb

```diff
@@ -691,22 +705,47 @@ def log_direct_schedule_submission_errors(e)
       end
 
       def set_type(appointment)
-        type = APPOINTMENT_TYPES[:request] if appointment[:kind] != 'cc' && appointment[:request_periods].present?
-
-        type ||= case appointment[:kind]
-                 when 'cc'
-                   if appointment[:start]
-                     APPOINTMENT_TYPES[:cc_appointment]
-                   else
-                     APPOINTMENT_TYPES[:cc_request]
-                   end
-                 else
-                   APPOINTMENT_TYPES[:va]
-                 end
+        type = if cerner?(appointment)
+                 cerner_type(appointment)
+               else
+                 non_cerner_type(appointment)
+               end
 
         appointment[:type] = type
       end
 
+      # Determines the type of appointment for Cerner appointments.
+      # @param appointment [Hash] the appointment to determine the type for
+      #
+      # @return [String] the type of appointment
+      #
+      def cerner_type(appointment)
+        if appointment[:end].present?
```

Comment:
I'm not too familiar with cerner type appointment so I can't say if this is the right field to check. Would be nice for someone more familiar to double check cc @simiadebowale but it at least sounds reasonable.